openvdb/points/AttributeArray.h

```diff
@@ -836,6 +1019,92 @@ UnitVecCodec::encode(const math::Vec3<T>& val, StorageType& data)
 }
 
 
+////////////////////////////////////////
+
+// AttributeArray implementation
+
+#if OPENVDB_ABI_VERSION_NUMBER >= 6
+template <typename IterT>
+void AttributeArray::copyValuesUnsafe(const AttributeArray& sourceArray, const IterT& iter,
+    bool rangeChecking/* = false*/)
+{
+    // ensure both arrays have float-float or integer-integer value types
+    assert(sourceArray.valueTypeIsFloatingPoint() != this->valueTypeIsFloatingPoint());
```

Comment:
Is this assert correct?

Reply:
No, that doesn't look correct, I'll fix that. Thanks.

Reply:
We compile a debug build, however we don't execute it because the time to do so is prohibitively slow. Ideally we should have an optimized build with asserts switched on which does get executed to catch these cases, but that's probably something to factor into the migration to the new Jenkins CI at this stage.